instrumentation/pulsar/pulsar-2.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/ProducerImplInstrumentation.java

```diff
@@ -82,54 +78,9 @@ public static void before(
       }
 
       Context context = producerInstrumenter().start(parent, request);
-      callback = new SendCallbackWrapper(context, request, callback);
```

Comment:
An alternative would have been to replace `SendCallbackWrapper` with a jdk proxy to handle the change in `sendComplete` signature. What you chose to do is fine too.